### PR TITLE
Improve code & complete activity when creating comments

### PR DESCRIPTION
**Pull request by Betree**
Part of https://github.com/opencollective/opencollective/issues/5887

- Add missing `HostCollectiveId` to the activity, make the call synchronous 
- Deprecate `ConversationId` in favor of `conversation` (reference)
- Improve overall code now that we don't need to support GQLV1

### CODE REVIEW

File: server/graphql/v2/input/ConversationReferenceInput.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * An input for referencing Conversations
+ */
+export const ConversationReferenceInput = new GraphQLInputObjectType({
+  name: 'ConversationReferenceInput',
+  fields: () => ({
+    id: {
+      type: GraphQLString,
+      description: 'The public id identifying the update',
```
Comment (SudharakaP) [inline server/graphql/v2/input/ConversationReferenceInput.ts:15]:
```suggestion
      description: 'The public id identifying the conversation',
```

File: server/graphql/v2/input/ConversationReferenceInput.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return idDecode(input['id'], IDENTIFIER_TYPES.UPDATE);
+  } else if (input['legacyId']) {
+    return input['legacyId'];
+  } else {
+    return null;
+  }
+};
+
+/**
+ * Retrieve an expense from an `ExpenseReferenceInput`
```
Comment (SudharakaP) [inline server/graphql/v2/input/ConversationReferenceInput.ts:34]:
```suggestion
 * Retrieve an conversation from an `ConversationReferenceInput`
```

File: server/graphql/v2/input/ConversationReferenceInput.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ */
+export const fetchConversationWithReference = async (input, { loaders = null, throwIfMissing = false } = {}) => {
```
Comment (SudharakaP) [inline server/graphql/v2/input/ConversationReferenceInput.ts:36]: This function is not used anywhere? 🤔

**Review by SudharakaP — APPROVED** [review]

File: server/graphql/v2/input/ConversationReferenceInput.ts (hunk 2 above)
Comment (znarf) [reply server/graphql/v2/input/ConversationReferenceInput.ts:34]: a conversation

**Comment by znarf** [comment]
Looks good!

File: server/graphql/v2/input/ConversationReferenceInput.ts (hunk 3 above)
Comment (Betree) [reply server/graphql/v2/input/ConversationReferenceInput.ts:36]: Not yet, but it's there to match the pattern we have in all reference helpers: `fetch*WithReference`. Will be useful when we update the `conversation` query.

**Pull request merged**